assets/app/scripts/controllers/pod.js

```diff
@@ -79,10 +81,35 @@ angular.module('openshiftConsole')
       watches.push(DataService.watch("builds", $scope, function(builds) {
         $scope.builds = builds.by("metadata.name");
         Logger.log("builds (subscribe)", $scope.builds);
-      }));      
+      }));
+
+      (function initLogs() {
+        angular.extend($scope, {
+          logs: [],
+          logsLoading: true,
+          canShowDownload: false,
+          initLogs: initLogs
+        });
+        DataService.watchLogStream("pods/log", $routeParams.pod, $scope, function(line) {
+          $scope.$apply(function() {
+            $scope.logs.push({text: line});
+            $scope.canShowDownload = true;
+          });
+        }, function() {
+          $scope.$apply(function() {
+            $scope.logsLoading = false;
+          });
+        })
```

Comment:
Suggest a comment saying on message / on close for these callbacks so you don't have to open data.js to see what they mean.